packages/protocol/engine/congestioncontrol/congestioncontrol.go

```diff
@@ -1,14 +1,31 @@
 package congestioncontrol
 
 import (
+	"time"
+
+	"github.com/iotaledger/hive.go/core/generics/event"
 	"github.com/iotaledger/hive.go/core/generics/options"
+	"github.com/iotaledger/hive.go/core/identity"
+
+	"github.com/iotaledger/goshimmer/packages/protocol/engine/congestioncontrol/icca/scheduler"
+	"github.com/iotaledger/goshimmer/packages/protocol/engine/consensus/acceptance"
+	"github.com/iotaledger/goshimmer/packages/protocol/engine/tangle"
+	"github.com/iotaledger/goshimmer/packages/protocol/engine/tangle/models"
 )
 
 type CongestionControl struct {
+	Events *Events
+
+	optsScheduler []options.Option[scheduler.Scheduler]
+
+	*scheduler.Scheduler
 }
 
-func New(opts ...options.Option[CongestionControl]) (congestionControl *CongestionControl) {
+func New(isBlockAccepted func(models.BlockID) bool, blockAcceptedEvent *event.Linkable[*acceptance.Block, acceptance.Events, *acceptance.Events], tangle *tangle.Tangle, accessManaMapRetrieverFunc func() map[identity.ID]float64, totalAccessManaRetrieveFunc func() float64, rate time.Duration, opts ...options.Option[CongestionControl]) (congestionControl *CongestionControl) {
```

Comment:
> This seems confusing as we usually talk about BPS. Could we change this as well?
I talked to Andrew about this and this will need to be converted to Work Units Per Second ultimately. Let's leave it as it is for now as not to change it twice.